src/client.rs

```diff
@@ -1010,14 +1010,14 @@ where
                 // SET SHARD TO
                 Some((Command::SetShard, _)) => {
                     // Selected shard is not configured.
-                    if query_router.shard() >= pool.shards() {
+                    if query_router.shard().unwrap_or(0) >= pool.shards() {
```

Comment:
This should never be None with the current query router implementation but to be safe this should raise a client error if it is None.

Reply:
I agree.